ESP8266/ESP8266.cpp

```diff
@@ -623,13 +624,16 @@ int32_t ESP8266::_recv_tcp_passive(int id, void *data, uint32_t amount, uint32_t
 {
     int32_t ret;
 
+    _smutex.lock();
+
+    _process_oob(timeout, true);
+
     // return immediately if no data is available
-    if (_sock_i[id].tcp_data_avbl == 0) {
+    if (_sock_i[id].tcp_data_avbl == 0 && _sock_i[id].open) {
```

Comment:
Do want to send the command when the socket is closed? In this case I'd wrap the command send sequence in `if (_sock_i[id].tcp_data_avbl != 0)`, and if you skip it, just go to the common handling with `ret` being set to would block